tests/src/common.hh

```diff
@@ -121,3 +128,65 @@ template <typename NonLinearOperator>
       << "The Jacobian of calculateMatrix is not the Jacobian of calculateVector." << messageIfFailed;
   return t;
 }
+
+template <typename NonLinearOperator, typename FiniteElement>
+[[nodiscard]] auto checkCauchyStressOf2DElement(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
```

Comment:
This function currently only worsk for 2D quadrilaterals with Linear Lagrange ansatz functions. Can you write a static_Assert for the dimension and a assert for the geometry type?

Something like
```cpp
static_assert(FiniteElement::Traits::mydim == 2,"Message"); //checks the size of the element at compile time
assert(fe.localView().element().type() == Dune::GeometryTypes::quadrilateral);  //checks the reference element type of the element at run time. 

static_assert(Check that basis is first order lagrange); // we can do that together, since this is a bit ugly
```